aws/cluster/cluster-spec.tf

```diff
@@ -287,3 +290,26 @@ locals {
     }, length(var.minion-additional-sgs) > 0 ? { additionalSecurityGroups = var.minion-additional-sgs } : {})
   }
 }
+
+data "aws_iam_policy_document" "master-additional" {
+  source_json = var.master-additional-policies
+
+  statement {
+    actions = [
+      "s3:GetObject",
+    ]
+    resources = [
+      "arn:aws:s3:::${var.kops-state-bucket}/terraform-addons/${var.cluster-name}/*"
```

Comment:
Any reason to use a top-level `terraform-addons/${var.cluster-name}/` as opposed to `${var.cluster-name}/terraform-addons/` which would mean you probably don't need this policy althogether?
`s3://${var.kops-state-bucket}/${var.cluster-name}/terraform-addons/` in other words? 
Its quite tightly coupled?

Reply:
I had your proposed version implemented originally, but kops refuses to destroy cluster if there are files on that path that it didn’t create, which is why I have changed the directory structure.